pyccel/parser/syntactic.py

```diff
@@ -774,10 +775,18 @@ def fill_types(ls):
         if 'private' in decorators.keys():
             is_private = True
 
+        if 'inline' in decorators.keys():
+            is_inline = True
+
         body = CodeBlock(body)
 
-        returns = [i.expr for i in body.get_attribute_nodes(Return, excluded_nodes = (Assign, FunctionCall, PyccelInternalFunction))]
+        returns = [i.expr for i in body.get_attribute_nodes(Return,
+                    excluded_nodes = (Assign, FunctionCall, PyccelInternalFunction, FunctionDef))]
         assert all(len(i) == len(returns[0]) for i in returns)
+        if is_inline and len(returns)>1:
+            errors.report("Inline functions cannot have multiple return statements",
+                    symbol = stmt,
+                    severity = 'error')
```

Comment:
Because it is very hard to translate all cases. E.g:
```
@inline
def f(n : int):
    if n == 0:
        return 0
    # calculations
    return x
```

Reply:
I think it would be nice to handle this for cases like:
```
def f(a : int = None):
    if a is None:
        return 1
    else:
        return 2
```
but I think it is an extension that can be left for another PR if this is wanted as it requires inserting `else`s if the code is written as above

Reply:
I see. It is complicated to stop the calculations after each return statement.